popup.js

```diff
@@ -1,12 +1,79 @@
 import { effects } from "./effects.js";
 
+createWebsiteContainer().catch(console.error);
 
 createForm().catch(console.error);
 
 createMasterToggle().catch(console.error);
 
 createNewWebsiteListeners().catch(console.error);
 
+async function initializeMasterDelete(websites) {
+    const masterDeleteButton = document.getElementById('master-delete-button');
+
+    // When the master delete button is pressed
+    masterDeleteButton.addEventListener('click', async () => {
+        console.log("master delete pressed");
+        let websites = await getWebsites();
+        let checkboxes = document.querySelectorAll('.websites:checked');
+        let indexesToDelete = Array.from(checkboxes, checkbox => parseInt(checkbox.id.replace('website-', '')));
+        console.log(indexesToDelete);
+
+        indexesToDelete.sort((a, b) => b - a); // Sort in descending order
+        indexesToDelete.forEach(index => {
+            websites.splice(index, 1);
+        });
+
+        await saveWebsites(websites);
+        createWebsiteContainer().catch(console.error);
+    });
+}
+
+
+async function createWebsiteContainer() {
+
+    getWebsites().then(websites => {
+        console.log(websites);
+        // Select the container where you want to append the websites
+        let container = document.getElementById('container-websites');
+
+        // Add delete button for checked websites
+
+        // Clears container of stale websites
+        container.innerHTML = '';
+
+        websites.forEach((website, index) => {
+            let div = document.createElement('div');
+            let input = document.createElement('input');
+            let label = document.createElement('label');
+            let span = document.createElement('span');
+            let button = document.createElement('button');
+            // Add individual delete buttons for each website
+
+            input.type = 'checkbox';
+            input.id = `website-${index}`;
+            input.className = "websites";
+            span.textContent = website;
+
+            button.textContent = "Delete";
+            button.className = "delete-button";
+
+            button.addEventListener('click', async () => {
+                websites.splice(index, 1);
+                await saveWebsites(websites);
+                createWebsiteContainer().catch(console.error);
+            });
+
+            label.appendChild(input);
+            label.appendChild(span);
+            div.appendChild(label);
+            div.appendChild(button);
+            container.appendChild(div);
+        });
+        initializeMasterDelete(websites).catch(console.error);
+    }).catch(console.error);
```

Comment:
Hey @IanMorriso , is the syntax for all of the .catch() calls correct? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

I thought it was something like .catch(error) {
console.error(error);
}